### PR TITLE
Adding React quiz

### PR DESCRIPTION
**Pull request by gedas-monginas**
I can't assign it, but if I could it would be @twhid

### CODE REVIEW

**Comment by twhid** [comment]
@gedas-monginas You can't assign at all or just this repo?

**Comment by gedas-monginas** [comment]
nope, can't change any of the settings: label, milestone, assignee

File: react/.npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+registry=https://registry.npmjs.org/
```
Comment (twhid) [inline react/.npmrc]: This file isn't necessary.

File: react/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "version": "1.0.0",
+    "description": "1stdibs Front-end quiz based on React, Redux, Node",
+    "main": "index.js",
+    "scripts": {
+        "cleaninstall": "rm -rf ./node_modules && npm cache clean && npm install",
+        "dev": "node server/index.js"
+    },
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/kostasmanionis/fng-quiz.git"
```
Comment (twhid) [inline react/package.json]: We'll want to fix this before we merge.

**Comment by twhid** [comment]
Sorry, but I completely disagree with @antialias's comment: 

> Testing a candidate's knowledge of react is less important than knowing if they can get a project up and running from scratch.

Getting a bunch of boilerplate code together to actually *start* building your app seems less important than being able to write the application. There are relatively few times that a FE dev at 1stdibs will need to do this and we'll provide generators/scaffolding to help them.

**Comment by twhid** [comment]
Having said that, we could make it more clear in the readme as to this quiz is testing, ie client-side react views and state management.

**Comment by antialias** [comment]
"more important" might have been a bit stronger than necessary, but put another way, would you trust a candidate's abilities if they _weren't_ able to put together a small web application from scratch? I wouldn't.

**Comment by antialias** [comment]
> There are relatively few times that a FE dev at 1stdibs will need to do this and we'll provide generators/scaffolding to help them.

Also, someone has to make the scaffolding. Again, I'm backing off on my earlier "more important than react" comment since it's an apple vs oranges thing, _but_ setting up a project is an important skill to have, and do like how the current front-end quiz allows the candidate to give insights into how they set up their projects. 

For example, I would give bonus points to someone if they experimented with bundling with rollup.js instead of webpack.

**Comment by twhid** [comment]
I've become ambivalent.

[1] On one hand, if the test were much simpler -- more like the Backbone test -- it could give us insight into how the candidate makes tech decisions. Did they decide to go with the simplest set-up possible? Or, did they go all out (as the bootstrapping in this PR does)?

[2] On the other, will that tell us that much? The candidate could decide that we're focused on the actual app, and slap together a [quicky React environment](https://facebook.github.io/react/docs/tutorial.html#getting-started) .. should that count against them? I'm not so sure. If we leave it this PR as-is, a lot of that mystery goes away, we're only looking at the actual React app that they've written within the highly controlled bootstrapped environment.

I'm leaning towards 2 as a better test.

File: react/readme.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+**Pdp:**
+
+- Fetch item data from server side using this endpoint: `[GET] /pdp/{id}/data`
+- Render item, example [layout](./examples/pdp.png)
+
+**Bonus points:**
+
+- Add item favoriting:
+  + Item can be added and removed to/from favorites from both pages by clicking on the `heart` icon
+  + Its up to you were you store the data about favourited items (server, db, browser, file or your own solution)
```
Comment (hellatan) [inline react/readme.md]: it should be spelled `favorited` :stuck_out_tongue:

**Comment by gedas-monginas** [comment]
all comments addressed

**Pull request merged**